kafka.go

```diff
@@ -282,27 +286,69 @@ func (client *KafkaClient) Register(subscribeBuilder *SubscribeBuilder) error {
 	config := client.subscribeConfig
 	config.Topic = topic
 	config.GroupID = groupID
-	config.StartOffset = kafka.LastOffset
+	config.StartOffset = subscribeBuilder.offset
 	config.MaxWait = kafkaMaxWait
 	reader := kafka.NewReader(config)
 
 	go func() {
-		defer reader.Close() // nolint: errcheck
+		var eventProcessingFailed bool
+
+		defer func() {
+			if eventProcessingFailed {
+				if subscribeBuilder.ctx.Err() != nil {
+					// the subscription is shutting down. triggered by an external context cancellation
```

Comment:
Shouldn't we write to the log what happen here? Just for debugging. Thoughts?

Reply:
@ajiyakin improved